### ISSUE TITLE
Handle manual deletion of resources

### ISSUE THREAD

**Issue by mikkoc** [onelogin/terraform-provider-onelogin#59]
Currently we find that Terraform plan breaks very very badly when a resource is deleted manually from OneLogin, i.e. an app or a role_attachment.

It would be nice if the provider could handle this: detect that a resource in the tfstate is not there anymore, and re-create it.

**Comment by dcaponi** [comment]
@mikkoc I'm actually not a seasoned terraform user myself, so I may be wrong, but does terraform handle this scenario with other providers? My understanding is that once you go in on terraform, its not advised to manually manage resources outside of terraform.

I did however anticipate this being a thing, and I wrote up a [CLI tool](https://github.com/onelogin/onelogin) that might help.

If you install that, and run `onelogin terraform-import onelogin_apps` that should bring your terraform plan back in line with what's actually in your onelogin account.

Please give it a try on a copy of your terraform plan first as the terraform import tool is somewhat experimental in nature.

**Comment by dcaponi** [comment]
Ah ok I see. Rather than the remote (AWS, Azure, OneLogin) being the source of truth, you want .tfstate to be the source of truth and to recreate stuff if it was manually removed; and you're saying this is a pattern with other providers you used in the past. 

Sound about right?

**Comment by mikkoc** [comment]
that is absolutely correct, yes. If Terraform detects that a resource is in the state file, but not present in the origin, it will try to re-create it (unless, obviously, the code that created that resource in the first place is also deleted).

**Comment by Subterrane** [comment]
I know it's been 4 years but I believe the recent updates have fixed this.
